plugin/core/sessions.py

```diff
@@ -1677,14 +1680,20 @@ def run_async() -> None:
 
             sublime.set_timeout_async(run_async)
             return Promise.resolve(None)
+        self._is_executing_refactoring_command = is_refactoring
         # TODO: Our Promise class should be able to handle errors/exceptions
-        return Promise(
+        execute_command = Promise(
             lambda resolve: self.send_request(
                 Request("workspace/executeCommand", command, None, progress),
                 resolve,
                 lambda err: resolve(Error(err["code"], err["message"], err.get("data")))
             )
         )
+        execute_command.then(lambda _: self._reset_is_executing_refactoring_command())
+        return execute_command
+
+    def _reset_is_executing_refactoring_command(self) -> None:
+        self._is_executing_refactoring_command = False
```

Comment:
```suggestion
        execute_command.then(lambda _: self._reset_is_executing_refactoring_command(is_refactoring))
        return execute_command

    def _reset_is_executing_refactoring_command(self, is_refactoring: bool) -> None:
        if is_refactoring:
            self._is_executing_refactoring_command = False
```

Only need to reset if this value was activated by this command (see comment above).